File: Scadex.Model/Dtos/Permission/Queries/PermissionDto.cs
using Scadex.Core.Model;

namespace Scadex.Model.Dtos.Permission.Queries;

public class PermissionDto : IDto
{
    public int Id { get; set; }
    public string Code { get; set; } = null!;
    public string DisplayName { get; set; } = null!;
    public string Category { get; set; } = null!;
    public string? CreatedBy { get; set; }
    public string? UpdatedBy { get; set; }
    public DateTime? CreateDateUtc { get; set; }
    public DateTime? UpdateDateUtc { get; set; }
}

File: Scadex.WebUI/src/models/permission/queries/permissionDto.ts
/** Ayna: Scadex.Model/Dtos/Permission/Queries/PermissionDto.cs — seed'li, salt okunur katalog. */
export interface PermissionDto {
  id: number;
  /** `EntityEnums.Permission` üyesinin adı; token'daki `permission` claim'i budur. */
  code: string;
  displayName: string;
  category: string;
  createdBy: string | null;
  updatedBy: string | null;
  createDateUtc: string | null;
  updateDateUtc: string | null;
}
